packages/ember-data/tests/integration/relationships/has_many_test.js

```diff
@@ -144,6 +144,52 @@ test("A serializer can materialize a hasMany as an opaque token that can be lazi
   });
 });
 
+test("If has-many options buildUrl is set, the adapter's buildURLForHasMany is called and the result is passed to findHasMany", function() {
```

Comment:
Can you add a test that assert that if buildURLForHasMany is provided, and user provided a URL, we always go to the user provided url and not generate a url